test/it/controller/etcdopstask/helper.go

```diff
@@ -0,0 +1,130 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package etcdopstask
+
+import (
+	"context"
+	"net/http"
+	"testing"
+	"time"
+
+	druidconfigv1alpha1 "github.com/gardener/etcd-druid/api/config/v1alpha1"
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/core/v1alpha1"
+	"github.com/gardener/etcd-druid/internal/controller/etcdopstask"
+	"github.com/gardener/etcd-druid/internal/controller/etcdopstask/handler"
+	"github.com/gardener/etcd-druid/internal/controller/etcdopstask/handler/ondemandsnapshot"
+	"github.com/gardener/etcd-druid/test/it/setup"
+	testutils "github.com/gardener/etcd-druid/test/utils"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	. "github.com/onsi/gomega"
+)
+
+// ReconcilerTestEnv encapsulates the test environment required to test the EtcdOpsTask reconciler.
+type ReconcilerTestEnv struct {
+	itTestEnv  setup.DruidTestEnvironment
+	reconciler *etcdopstask.Reconciler
+}
+
+// InjectHTTPClient injects an HTTP client into the reconciler's task handler registry for testing.
+func (env *ReconcilerTestEnv) InjectHTTPClient(httpClient *http.Client) {
+	registry := env.reconciler.GetTaskHandlerRegistry()
+	registry.Register("OnDemandSnapshot", func(k8sclient client.Client, task *druidv1alpha1.EtcdOpsTask, _ *http.Client) (handler.Handler, error) {
+		return ondemandsnapshot.New(k8sclient, task, httpClient)
+	})
+}
+
+// initializeEtcdOpsTaskReconcilerTestEnv sets up the test environment
+func initializeEtcdOpsTaskReconcilerTestEnv(t *testing.T, itTestEnv setup.DruidTestEnvironment, clientBuilder *testutils.TestClientBuilder) ReconcilerTestEnv {
+	var g *WithT
+	if t != nil {
+		g = NewWithT(t)
+	} else {
+		g = NewWithT(&testing.T{})
+	}
+	var (
+		reconciler *etcdopstask.Reconciler
+	)
+
+	g.Expect(itTestEnv.CreateManager(clientBuilder)).To(Succeed())
+
+	itTestEnv.RegisterReconciler(func(mgr manager.Manager) {
+		reconciler = etcdopstask.NewReconciler(mgr, &druidconfigv1alpha1.EtcdOpsTaskControllerConfiguration{
+			ConcurrentSyncs: ptr.To(3),
+			RequeueInterval: &metav1.Duration{Duration: 15 * time.Second},
+		})
+		g.Expect(reconciler.RegisterWithManager(mgr)).To(Succeed())
+	})
+
+	g.Expect(itTestEnv.StartManager()).To(Succeed())
+	if t != nil {
+		t.Log("successfully registered etcdopstask reconciler with manager and started manager")
+	}
+
+	return ReconcilerTestEnv{
+		itTestEnv:  itTestEnv,
+		reconciler: reconciler,
+	}
+}
+
+// newTestTask creates a new EtcdOpsTask instance.
+func newTestTask(namespace string, state *druidv1alpha1.TaskState) *druidv1alpha1.EtcdOpsTask {
```

Comment:
Create a new package under `etcdopstask` for ondemand snapshot. In that package have all helper functions specific to ondemand snapshot and IT tests.
When more ops tasks are added they can create their own packages. All common helper functions should be kept directly under `etcdopstask` package.

Ideally you should create builders for each ops task and that should be under test/utils package.